obsidian.d.ts

```diff
@@ -290,307 +1583,805 @@ export abstract class AbstractInputSuggest<T> extends PopoverSuggest<T> {
      * @public
      */
     constructor(inputEl: T);
     /**
+     * Sets the disabled state of the input element.
+     *
+     * @param disabled - Whether to disable the input element.
+     * @returns The text component.
+     *
+     * @example
+     * ```ts
+     * textComponent.setDisabled(true);
+     * ```
+     *
      * @public
      */
     setDisabled(disabled: boolean): this;
     /**
+     * Gets the value of the input element.
+     *
+     * @returns The value of the input element.
+     *
+     * @example
+     * ```ts
+     * const value = textComponent.getValue();
+     * ```
+     *
      * @public
      */
     getValue(): string;
     /**
+     * Sets the value of the input element.
+     *
+     * @param value - The value to set.
+     * @returns The text component.
+     *
+     * @example
+     * ```ts
+     * textComponent.setValue('foo');
+     * ```
+     *
      * @public
      */
     setValue(value: string): this;
     /**
+     * Sets the placeholder of the input element.
+     *
+     * @param placeholder - The placeholder to set.
+     * @returns The text component.
+     *
+     * @example
+     * ```ts
+     * textComponent.setPlaceholder('foo');
+     * ```
+     *
      * @public
      */
     setPlaceholder(placeholder: string): this;
     /**
+     * Manually invokes the callback registered with `onChange`.
+     *
+     * @example
+     * ```ts
+     * textComponent.onChange((value) => {
+     *     console.log(value);
+     * });
+     * textComponent.onChanged();
+     * ```
+     *
      * @public
      */
     onChanged(): void;
     /**
+     * Sets the callback to handle when the value of the input element changes.
+     *
+     * @param callback - The callback to handle when the value of the input element changes.
+     * @returns The text component.
+     *
+     * @example
+     * ```ts
+     * textComponent.onChange((value) => {
+     *     console.log(value);
+     * });
+     * ```
+     *
      * @public
      */
     onChange(callback: (value: string) => any): this;
 }
 
 /**
  * Adds an icon to the library.
+ *
  * @param iconId - the icon ID
  * @param svgContent - the content of the SVG.
+ *
+ * @example
+ * ```ts
+ * addIcon('icon-id', '<svg>...</svg>');
+ * ```
+ *
  * @public
  */
 export function addIcon(iconId: string, svgContent: string): void;
 
 /**
  * This is the API version of the app, which follows the release cycle of the desktop app.
  * Example: '0.13.21'
+ *
  * @public
  */
 export let apiVersion: string;
 
 /**
+ * The main app object.
+ *
  * @public
  */
 export class App {
 
-    /** @public */
+    /**
+     * The keymap object.
+     *
+     * @public
+     */
     keymap: Keymap;
-    /** @public */
+    /**
+     * The scope object.
+     *
+     * @public
+     */
     scope: Scope;
 
-    /** @public */
+    /**
+     * The workspace object.
+     *
+     * @public
+     */
     workspace: Workspace;
 
-    /** @public */
+    /**
+     * The vault object.
```

Comment:
I don't see the point in documenting this, it's just repeating what's already visible from the types alone.